neat_cache/lib/src/providers/resp.dart

```diff
@@ -215,13 +215,39 @@ class RespClient {
       }
     }
 
+    // Ensure that we also attempt to close the [Socket] after all pending
+    // commands have been sent.
+    //
+    // This may be necessary, as it appears that [Socket] from 'dart:io' may
+    // throw [StateError], if it's bound to stream when [close] is called.
+    //
+    // See: https://github.com/dart-lang/sdk/issues/41707
+    _pendingStream = _pendingStream.then((_) async {
+      try {
+        await _output.close();
+      } catch (e) {
+        // ignore
+      }
+    });
+
     if (!force) {
       scheduleMicrotask(() async {
-        await _output.close().catchError((_) {/* ignore */});
+        try {
+          await _output.close();
+        } catch (e) {
+          // ignore
+        }
       });
       await _closed.future;
     } else {
-      await _output.close().catchError((_) {/* ignore */});
+      try {
+        // NOTE: Sadly, this can throw, it can even throw synchroniously,
+        //       implying [Socket] from 'dart:io' fails to implement
+        //       [StreamSink] as specified.
```

Comment:
```suggestion
        // NOTE: Sadly, this can throw, it can even throw synchroniously,
        //       implying [Socket] from 'dart:io' fails to implement
        //       [StreamSink] as specified.
```
If this is true we should add a issue number here.